evm_arithmetization/src/prover.rs

```diff
@@ -170,6 +202,427 @@ where
+
+    let auxiliary_columns = all_auxiliary_columns::<F, C, D>(
+        all_stark,
+        config,
+        &trace_poly_values,
+        &ctl_data_per_table,
+        &ctl_challenges,
+    );
+
+    // We compute the Field Merkle Tree of all auxiliary columns.
+    let auxiliary_columns_sorted: Vec<_> = Table::all_sorted()
+        .iter()
+        .map(|&table| auxiliary_columns[*table].clone())
+        .collect();
+    let auxiliary_columns_sorted_flat = auxiliary_columns_sorted
+        .clone()
+        .into_iter()
+        .flatten()
+        .collect();
+    let auxiliary_commitment = timed!(
+        timing,
+        "compute auxiliary commitments",
+        BatchFriOracle::<F, C, D>::from_values(
+            auxiliary_columns_sorted_flat,
+            rate_bits,
+            false,
+            cap_height,
+            timing,
+            &[None; NUM_TABLES],
+        )
+    );
+    challenger.observe_cap(&auxiliary_commitment.batch_merkle_tree.cap);
+
+    // Quotient polynomials.
+    // TODO: Implement it.
+    let alphas = challenger.get_n_challenges(config.num_challenges);
+    let quotient_polys = all_quotient_polys(
+        all_stark,
+        &trace_commitment,
+        &auxiliary_commitment,
+        None,
+        &ctl_data_per_table,
+        alphas,
+        config,
+    );
+
+    // We compute the Field Merkle Tree of all quotient polynomials.
+    let quotient_polys_sorted: Vec<_> = Table::all_sorted()
+        .iter()
+        .map(|&table| quotient_polys[*table].clone())
+        .collect();
+    let quotient_polys_sorted_flat = quotient_polys_sorted
+        .clone()
+        .into_iter()
+        .flatten()
+        .collect();
+    let quotient_commitment = timed!(
+        timing,
+        "compute quotient commitments",
+        BatchFriOracle::<F, C, D>::from_values(
+            quotient_polys_sorted_flat,
+            rate_bits,
+            false,
+            cap_height,
+            timing,
+            &[None; NUM_TABLES],
+        )
+    );
+    challenger.observe_cap(&quotient_commitment.batch_merkle_tree.cap);
+
+    let zeta = challenger.get_extension_challenge::<D>();
+
+    // To avoid leaking witness data, we want to ensure that our opening locations,
+    // `zeta` and `g * zeta`, are not in our subgroup `H`. It suffices to check
+    // `zeta` only, since `(g * zeta)^n = zeta^n`, where `n` is the order of
+    // `g`.
+    let degree_bits = trace_commitment.degree_bits[0];
+    let g = F::primitive_root_of_unity(degree_bits);
+    ensure!(
+        zeta.exp_power_of_2(degree_bits) != F::Extension::ONE,
+        "Opening point is in the subgroup."
+    );
+
+    let mut all_fri_instances: Vec<FriInstanceInfo<F, D>> = Vec::new();
+
+    // Get the FRI openings and observe them.
+    // Compute all openings: evaluate all committed polynomials at `zeta` and, when
+    // necessary, at `g * zeta`.
+    // TODO: Need batched openings.
+    let openings = StarkOpeningSet {
```

Comment:
The `StarkOpeningSet` is not suited for batch openings. We would need either to create a new struct, or have e.g. a `Vec<StarkOpeningSet>`.